src/runtime/pickle_runner.ts

```diff
@@ -375,9 +411,36 @@ export default class PickleRunner {
         },
       })
     }
-    return await this.invokeStep(
-      testStep.pickleStep,
-      testStep.stepDefinitions[0]
+
+    const beforeStepHooksResult = await this.runStepHooks(
```

Comment:
Can we rename this from `beforeStepHooksResult` to `beforeStepHookResults`? I think moving the `s` to the end make it easier to tell this is an array.

Same for `afterStepHooksResult` -> `afterStepHookResults`.